bahmnicore-omod/src/main/resources/liquibase.xml

```diff
@@ -4658,13 +4673,143 @@
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+                select count(*) from concept_reference_map rmap
+                join concept_reference_term rterm on rmap.concept_reference_term_id = rterm.concept_reference_term_id
+                and rterm.code='Non-Coded Diagnosis'
+                join concept_reference_source rsource on rsource.concept_source_id = rterm.concept_source_id and rsource.name
+                = 'org.openmrs.module.emrapi'
+                join concept c on c.concept_id = rmap.concept_id and c.version is null ;
+            </sqlCheck>
+        </preConditions>
+        <comment>Retire non Bahmni Diagnosis concept with reference mapping code 'Non-Coded Diagnosis'</comment>
+        <sql>
+            SET @concept_source_id = 0;
+            SET @concept_reference_term_id = 0;
+            SET @concept_id = 0;
+
+            SELECT concept_source_id into @concept_source_id from concept_reference_source where name = 'org.openmrs.module.emrapi';
+            SELECT concept_reference_term_id into @concept_reference_term_id from concept_reference_term where concept_source_id = @concept_source_id and code = 'Non-Coded Diagnosis';
+            SELECT concept_id into @concept_id from concept_reference_map where concept_reference_term_id = @concept_reference_term_id order by date_created desc limit 1;
+
+            UPDATE concept set retired = TRUE where version is null and concept_id = @concept_id;
+        </sql>
+    </changeSet>
+
+    <changeSet id="bahmni-emrapi-202208181249" author="BAHMNI" runAlways="true">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+                select count(*) from concept_reference_map rmap
+                join concept_reference_term rterm on rmap.concept_reference_term_id = rterm.concept_reference_term_id
+                and rterm.code='Diagnosis Certainty'
+                join concept_reference_source rsource on rsource.concept_source_id = rterm.concept_source_id and rsource.name
+                = 'org.openmrs.module.emrapi'
+                join concept c on c.concept_id = rmap.concept_id and c.version is null ;
+            </sqlCheck>
+        </preConditions>
+        <comment>Retire non Bahmni Diagnosis concept with reference mapping code 'Diagnosis Certainty'</comment>
+        <sql>
+            SET @concept_source_id = 0;
+            SET @concept_reference_term_id = 0;
+            SET @concept_id = 0;
+
+            SELECT concept_source_id into @concept_source_id from concept_reference_source where name = 'org.openmrs.module.emrapi';
+            SELECT concept_reference_term_id into @concept_reference_term_id from concept_reference_term where concept_source_id = @concept_source_id and code = 'Diagnosis Certainty';
+            SELECT concept_id into @concept_id from concept_reference_map where concept_reference_term_id = @concept_reference_term_id order by date_created desc limit 1;
```

Comment:
same as above. include relationship type and also version (not bahmni version)

Reply:
same as above